src/php/types/zval.rs

```diff
@@ -427,6 +427,16 @@ impl Zval {
         unsafe { zval_ptr_dtor(self) };
         self.u1.type_info = ty.bits();
     }
+
+    /// Extracts some type from a `Zval`.
+    ///
+    /// This is a wrapper function around `TryFrom`.
+    pub fn extract<T>(self) -> Result<T>
+    where
+        T: TryFrom<Self, Error = Error>,
```

Comment:
`TryFrom<Zval>` is not implemented for all types AFAIK, what about `T: FromZval<'a>` taking `&'a self` instead?

Or if you want to drop `self` then `for<'a> T: FromZval<'a>` and keep the function parameter as `self`. The `for<'a>` just makes it valid to have `T` as `String` and not `&str` (as it would reference the zval after dropping).